ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala

```diff
@@ -21,15 +26,19 @@ import scala.collection.mutable
 private[kvutils] trait CommitContext {
   private[this] val logger = LoggerFactory.getLogger(this.getClass)
 
-  def inputs: DamlStateMap
+  def inputsWithFingerprints: DamlStateMapWithFingerprints
+  final def inputs: DamlStateMap = inputsWithFingerprints.map {
```

Comment:
I wouldn't make this `final`. If we keep the same `CommitContext` interface for both pre-execution vs. normal use-cases then these two methods must be implemented in `Committer` in order to avoid rebuilding the input map every time for normal execution.
A simple solution is to use `.view` on whatever the inputs are and do the transformation to the other type (i.e., inputs with vs. without fingerprints).